File: solutions/haskell/darts/1/src/Darts.hs
module Darts (score) where

score :: Float -> Float -> Int
score x y 
   | distance x y > 10 = 0
   | distance x y <= 10 && distance x y > 5 = 1
   | distance x y > 1 && distance x y <= 5 = 5
   | distance x y <= 1 = 10
   | otherwise = 0
  where distance x y =  sqrt (x^2 + y^2 ) 

File: solutions/prolog/darts/1/darts.pl

score(X, Y, 10 ):- distance(X, Y, D), D =< 1.
score(X, Y, 5 ):- distance(X, Y, D), D > 1, D =< 5.
score(X, Y, 1 ):- distance(X, Y, D), D > 5, D =< 10.
score(X, Y, 0 ):- distance(X, Y, D), D > 10.

distance(X,Y, D):- Z is (X^2 + Y^2),
                     D is sqrt(Z).
